src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1384,12 +1383,30 @@ multiplicativeExpression
     :    unaryExpression ((STAR^ | DIV^ | MOD^ ) unaryExpression)*
     ;
 
+constantExpression
+    : IDENT
+    | constant
+    ;
+
+switchExpressionCasesGroup
+    :   ("case"^ (constantExpression) (COMMA constantExpression)*) LAMBDA^ expression SEMI
+    |   "default"^ LAMBDA^ statement
+    ;
+
+// Extended switch expression from Java12
+switchExpression
+    :   "switch"^ LPAREN expression RPAREN LCURLY
+             ( switchExpressionCasesGroup ) +
+        RCURLY
+    ;
+
 unaryExpression
     :    INC^ unaryExpression
     |    DEC^ unaryExpression
     |    MINUS^ {#MINUS.setType(UNARY_MINUS);} unaryExpression
     |    PLUS^  {#PLUS.setType(UNARY_PLUS);} unaryExpression
     |    unaryExpressionNotPlusMinus
+    |    switchExpression
```

Comment:
Just curious how you choose this section to add this to.

Reply:
I experimented with the JDK 12 compiler and it treated the switch expression as an unary expression, all the way down to incrementing and decrementing. So, for example, 
```java
return - switch (day) {
             case MON, FRI, SUN -> 6;
             case TUE -> 7;
             case THU, SAT -> 8;
             case WED -> 9;
         };
```

did work,

but
```java
return -- switch (day) {
             case MON, FRI, SUN -> 6;
             case TUE -> 7;
             case THU, SAT -> 8;
             case WED -> 9;
         };
```

did not.